src/cleanfigure.m

```diff
@@ -251,7 +246,7 @@ function pruneOutsideBox(meta, handle)
       id_remove  = id_remove(~idx);
   end
   % Replace the data points
-  replaceData(handle, id_replace, NaN(length(id_replace), size(data,2)));
```

Comment:
So the reason is, that we actually only insert NaNs, and i would have to handle the 3D case outside of the helper function as that defines the dimension of the NaN matrix.

The size of Data is always 2 btw as it is the projected data.